Tasks/AzureRmWebAppDeployment/AzureRMUtil.ts

```diff
@@ -0,0 +1,293 @@
+/// <reference path="../../definitions/node.d.ts" />
+/// <reference path="../../definitions/q.d.ts" />
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+
+var adal = require ('adal-node');
+var request = require ('request');
+var parseString = require('xml2js').parseString;
+
+import taskLib = require ('vsts-task-lib/task');
+import Q = require('q');
+
+var AuthenticationContext = adal.AuthenticationContext;
+
+function getAuthorizationToken ( SPN ) : Q.Promise <string> {
+
+	var deferred = Q.defer<string>();
+	var authorityUrl = 'https://login.windows.net/' + SPN.tenantID; 
+	var resource = 'https://management.azure.com/';
```

Comment:
Yes currently we will have them globally as constants when we add support to specific cloud we can fix there